### PR TITLE
Change keepers to use the default contract transmitter

### PR DESCRIPTION
**Pull request by ferglor**
As part of the LOOPP work, we discovered that the keepers service was using a different contract transmitter to the other services (the pipeline contract transmitter)

To enable the LOOPP work to progress for keepers, this PR switches the keeper service to using the default contract transmitter, and also removes the pipeline contract transmitter 

The integration tests were failing on this PR initially due to the fact that the gas limit for the automation service is pulled from a different function vs other services. To fix this, the contract transmitter constructor now accepts a gas limit as a param.

### CODE REVIEW

**Review by krehermann — COMMENTED** [review]
seems like this is going in the right direction, but given all the integration test failures, it seems like there are some subtleties in the pipelineContractTransmitter that are being captured in the refactor.

File: core/services/relay/evm/evm.go [hunk 1]
```diff
@@ -410,7 +408,6 @@ func newContractTransmitter(lggr logger.Logger, rargs commontypes.RelayArgs, tra
 		checker.CheckerType = txm.TransmitCheckerTypeSimulate
 	}
 
-	gasLimit := configWatcher.chain.Config().EVM().GasEstimator().LimitDefault()
```
Comment (infiloop2) [inline core/services/relay/evm/evm.go:413]: I see this is being passed from three call sites. Why not keep this as is and add logic for pluginGasLimit override (same logic as lines 464-471)

File: core/services/relay/evm/evm.go [hunk 2]
```diff
@@ -442,55 +439,6 @@ func newContractTransmitter(lggr logger.Logger, rargs commontypes.RelayArgs, tra
...
-	}
-	if pluginGasLimit != nil {
-		gasLimit = *pluginGasLimit
-	}
-
-	return NewOCRContractTransmitter(
-		configWatcher.contractAddress,
-		configWatcher.chain.Client(),
-		configWatcher.contractABI,
-		ocrcommon.NewPipelineTransmitter(
```
Comment (infiloop2) [inline core/services/relay/evm/evm.go:477]: this implementation can also likely be deleted

File: core/services/relay/evm/evm.go [hunk 3]
```diff
@@ -373,7 +371,7 @@ func newConfigProvider(lggr logger.Logger, chain legacyevm.Chain, opts *types.Re
 	return newConfigWatcher(lggr, aggregatorAddress, contractABI, offchainConfigDigester, cp, chain, relayConfig.FromBlock, opts.New), nil
 }
 
-func newContractTransmitter(lggr logger.Logger, rargs commontypes.RelayArgs, transmitterID string, configWatcher *configWatcher, ethKeystore keystore.Eth) (*contractTransmitter, error) {
+func newContractTransmitter(lggr logger.Logger, rargs commontypes.RelayArgs, transmitterID string, configWatcher *configWatcher, ethKeystore keystore.Eth, gasLimit uint32) (*contractTransmitter, error) {
```
Comment (krehermann) [inline core/services/relay/evm/evm.go:374]: i'd rather this this implemented with a new opts struct than a explicit argument.

type configTransmitterOpts struct {
configWatcher *configWatcher
// override the gas limit default provided in the config watcher
gasLimit uint32
}

func newContractTransmitter (..., opts configTransmitterOpts) {
...
gasLimit := configWatcher.chain.Config().EVM().GasEstimator().LimitDefault()
if opts.gasLimit != 0 {
	gasLimit = opt.gasLimit
}

}

all the call site need to be updated, but only the automation call site needs to explicitly set a gasLimit option.

this implementation seems more inline with the intention of passing a *configwatcher and deriving relevant options/configuration from it

File: core/services/ocr2/plugins/ocr2keeper/integration_21_test.go [hunk 4]
```diff
@@ -229,8 +226,6 @@ func TestIntegration_KeeperPluginLogUpkeep(t *testing.T) {
 	g.Eventually(listener, testutils.WaitTimeout(t), cltest.DBPollingInterval).Should(gomega.BeTrue())
 	done()
 
-	runs := checkPipelineRuns(t, nodes, 1)
```
Comment (infiloop2) [inline core/services/ocr2/plugins/ocr2keeper/integration_21_test.go:232]: Have we explored any alternate assertions for non pipeline transmitter? Or do we feel they are covered by existing assertions

**Review by infiloop2 — CHANGES_REQUESTED** [review]
Added some inline comments. Major ones
1. Change in override behaviour of gas limit config values
2. Ensuring we are not weakening our test suite

File: core/services/ocr2/plugins/ocr2keeper/integration_21_test.go (hunk 4 above)
Comment (ferglor) [reply core/services/ocr2/plugins/ocr2keeper/integration_21_test.go:232]: I wasn't able to see how the other services assert their transmits, so I just assumed the existing assertions would be enough, but I can look into this further

Comment (infiloop2) [reply core/services/ocr2/plugins/ocr2keeper/integration_21_test.go:232]: Verifying the upkeeps are actually performed i think should give similar coverage but I'm not sure that's done everywhere (e.g. the other place where this is removed)

**Review by jmank88 — DISMISSED** [review]

File: core/services/ocr2/plugins/ocr2keeper/integration_21_test.go [hunk 5]
```diff
@@ -247,17 +242,21 @@ func TestIntegration_KeeperPluginLogUpkeep(t *testing.T) {
...
 			backend.Commit()
 			time.Sleep(time.Millisecond * 10)
 		}
-		t.Logf("Mined %d blocks, waiting for logs to be recovered", dummyBlocks)
 
-		expectedPostRecover := runs + emits
-		waitPipelineRuns(t, nodes, expectedPostRecover, testutils.WaitTimeout(t), cltest.DBPollingInterval)
+		t.Logf("Mined %d blocks, waiting for logs to be recovered", dummyBlocks)
 
+		listener, done := listenPerformedN(t, backend, registry, ids, int64(beforeDummyBlocks), len(ids))
```
Comment (amirylm) [inline core/services/ocr2/plugins/ocr2keeper/integration_21_test.go:257]: shouldn't you use `expectedPostRecover := runs + emits` instead of `len(ids)`

Comment (ferglor) [reply core/services/ocr2/plugins/ocr2keeper/integration_21_test.go:257]: `runs` is no longer available since we no longer have `checkPipelineRuns`

**Review by krehermann — APPROVED** [review]

**Review by krehermann — APPROVED** [review]

File: core/services/relay/evm/evm.go [hunk 6]
```diff
@@ -390,6 +394,8 @@ func newContractTransmitter(lggr logger.Logger, rargs commontypes.RelayArgs, tra
 		return nil, pkgerrors.New("no sending keys provided")
 	}
 
+	configWatcher := opts.configWatcher
```
Comment (infiloop2) [inline core/services/relay/evm/evm.go:397]: nit. this seems like a required option (code below will fail if this is nil) so not sure if it made more sense as a parameter before (unlike pluginGasLimit for example)

**Review by infiloop2 — DISMISSED** [review]

**Review by krehermann — APPROVED** [review]

**Review by infiloop2 — APPROVED** [review]

**Pull request merged**